bytebufferpool/fixedsizebytebuffer.go

```diff
@@ -0,0 +1,58 @@
+package bytebufferpool
+
+import (
+	"io"
+)
+
+// MaxSize default max size for byte buffer
+var MaxSize = 1024 * 1024
+
+// FixedSizeByteBuffer provides fixed size byte buffer
+//
+// ByteBuffer may be used with functions appending data to the given []byte
+// slice.
+//
+// Use Get for obtaining an empty byte buffer.
+type FixedSizeByteBuffer struct {
+	// B is a byte buffer to use in append-like workloads.
+	B    []byte
+	Used int
```

Comment:
`Used`  shouldn't be public, which could be accidentally modified by user, add a Len() func for `FixedSizeByteBuffer ` instead